compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -108,11 +108,7 @@ fn coerce_mutbls<'tcx>(
     from_mutbl: hir::Mutability,
     to_mutbl: hir::Mutability,
 ) -> RelateResult<'tcx, ()> {
-    match (from_mutbl, to_mutbl) {
-        (hir::Mutability::Mut, hir::Mutability::Mut | hir::Mutability::Not)
-        | (hir::Mutability::Not, hir::Mutability::Not) => Ok(()),
-        (hir::Mutability::Not, hir::Mutability::Mut) => Err(TypeError::Mutability),
-    }
+    if from_mutbl > to_mutbl { Ok(()) } else { Err(TypeError::Mutability) }
```

Comment:
Should this be `>=`? What happens in the `(Not, Not)` case?

Reply:
Yes, already fixed 😅

Reply:
> [00:19] waffle is a compiler: ok lmao
> [00:19] waffle is a compiler: I accidentally restricted coercions to `&mut` -> `&` only
> [00:19] waffle is a compiler: so you can't coerce `&` to `&`

This made it so no function that passes `&` around can compile 👍🏻